encoding/ssz/query/ssz_info.go

```diff
@@ -40,26 +43,62 @@ func (info *sszInfo) Size() uint64 {
 		return info.fixedSize
 	}
 
-	// NOTE: Handle variable-sized types.
-	return 0
+	switch info.sszType {
+	case List:
+		length := info.listInfo.length
+		elementSize := info.listInfo.element.Size()
+
+		return length * elementSize
+
+	case Container:
+		size := info.fixedSize
+		for _, fieldInfo := range info.containerInfo.fields {
+			if !fieldInfo.sszInfo.isVariable {
+				continue
+			}
+
+			size += fieldInfo.sszInfo.Size()
+		}
+		return size
+
+	default:
+		// NOTE: Handle other variable-sized types.
+		return 0
+	}
 }
 
-func (info *sszInfo) ContainerInfo() (containerInfo, error) {
+func (info *sszInfo) ContainerInfo() (*containerInfo, error) {
 	if info == nil {
-		return nil, fmt.Errorf("sszInfo is nil")
+		return nil, errors.New("sszInfo is nil")
 	}
 
 	if info.sszType != Container {
 		return nil, fmt.Errorf("sszInfo is not a Container type, got %s", info.sszType)
 	}
 
 	if info.containerInfo == nil {
-		return nil, fmt.Errorf("sszInfo.containerInfo is nil")
+		return nil, errors.New("sszInfo.containerInfo is nil")
 	}
 
 	return info.containerInfo, nil
 }
 
+func (info *sszInfo) ListInfo() (*listInfo, error) {
+	if info == nil {
+		return nil, errors.New("sszInfo is nil")
+	}
+
+	if info.sszType != List {
+		return nil, fmt.Errorf("sszInfo is not a List type, got %s", info.sszType)
+	}
+
+	if info.listInfo == nil {
+		return nil, errors.New("sszInfo.listInfo is nil")
+	}
+
+	return info.listInfo, nil
```

Comment:
```suggestion
	return info.listInfo, nil
```
I think it's better to return the value no matter what it is, and let the caller decide how they want to handle the nil case